### PR TITLE
gh-105375: Improve error handling in _ctypes

### PR DESCRIPTION
**Pull request by erlend-aasland**
Prevent repeated PyLong_FromVoidPtr() from possibly overwriting the
current exception.

* Issue: gh-105375

### CODE REVIEW

File: Modules/_ctypes/callbacks.c [hunk 1]
```diff
@@ -479,13 +479,20 @@ long Call_GetClassObject(REFCLSID rclsid, REFIID riid, LPVOID *ppv)
...
+            Py_DECREF(py_rclsid);
+            PyErr_WriteUnraisable(context ? context : Py_None);
+            return E_FAIL;
+        }
         PyObject *py_ppv = PyLong_FromVoidPtr(ppv);
-        if (!py_rclsid || !py_riid || !py_ppv) {
-            Py_XDECREF(py_rclsid);
-            Py_XDECREF(py_riid);
-            Py_XDECREF(py_ppv);
-            Py_DECREF(func);
```
Comment (sobolevn) [inline Modules/_ctypes/callbacks.c:488]: What about `func`? It is not handled anymore.

Comment (erlend-aasland) [reply Modules/_ctypes/callbacks.c:488]: Good catch, thanks!

Comment (erlend-aasland) [reply Modules/_ctypes/callbacks.c:488]: Added decref's for error paths; the happy path is already taken care of.

**Review by sobolevn — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Argh, the refleaks in `test_import` and `test_peg_generator` are still there.

**Pull request merged**